KnowWhereGraph/kwg-deployment
# Add a barebones GraphDB-ES-NGINX-Prometheus deplpoyment

Fixes #2 #3 #9 
## Overview
This PR marks the start of a dockerized KnowWhereGraph deployment. Each service has its own folder with readme, configuration, and docker-compose file. The overall deployment is run by the root makefile.

## Features

_Local_ deployments of 

- NGINX
- GraphDB (restarts when it dies, fixing https://github.com/KnowWhereGraph/operations/issues/6)
- Elasticsearch
- Prometheus

I was never able to get phuzzy link built for docker-I'll have to add the new API & phuzzy replacement in a new PR.
This will also be tested on a new VM (through Open Nebula) in time...

## To Test

1. Clone the repo
2. Generate the nginx certs (see the nginx readme)
3. Optionally add a GraphDB license
4. Use the make commands to start the dev stack (see root level readme)
5. Go to https://localhost/graphdb (note the https)

<img width="1727" alt="Screenshot 2023-09-26 at 8 38 39 PM" src="https://github.com/KnowWhereGraph/kwg-deployment/assets/9289652/db32a097-aa64-4a60-8f6c-2a63065853f4">

<img width="1728" alt="Screenshot 2023-09-28 at 8 21 26 PM" src="https://github.com/KnowWhereGraph/kwg-deployment/assets/9289652/66a9a8d4-2e85-4e4f-b136-e3ec47ecd944">
